source/appModules/teamtalk5classic.py

```diff
@@ -0,0 +1,245 @@
+#: These are name sets per client version.
+#: Stored in reverse so current clients find the right list faster.
+_msgNames = {
+	"4.5": [
+		LBL_CONNECTED,
+		"sep",
+		LBL_PTT,
+		LBL_VOX,
+		LBL_VIDEO,
+		LBL_DESKTOP,
+		LBL_STREAM,
+		"sep",
+		LBL_MUTE,
+		LBL_RECORD,
+		"sep",
+		LBL_MSG,
+	],
+	"4.3": [
+		LBL_CONNECTED,
+		"sep",
+		LBL_PTT,
+		LBL_VOX,
+		LBL_VIDEO,
+		LBL_DESKTOP,
+		LBL_MUTE,
+		LBL_RECORD,
+		"sep",
+		LBL_MSG,
+	],
+	"4.2": [
+		LBL_CONNECTED,
+		"sep",
+		LBL_PTT,
+		LBL_VOX,
+		"sep",
+		LBL_MUTE,
+		LBL_RECORD,
+		"sep",
+		LBL_MSG,
+	],
+	"4.1": [
+		LBL_CONNECTED,
+		"sep",
+		LBL_PTT,
+		LBL_VOX,
+		"sep",
+		LBL_MUTE,
+		"sep",
+		LBL_MSG,
+	],
+}
+
+# window control IDs
+ID_CHATOUT = 1204
+ID_CHATIN = 1206
+ID_BANNEDLIST = 1326
+ID_UNBANNEDLIST = 1327
+ID_TOOLBAR = 59392
+ID_VU = 1004
+
+
+class TTToolBarInfo(object):
+	"""Info about one TeamTalk toolbar and its items.
+	"""
+
+	def __init__(self, NVDAObj=None):
+		if not NVDAObj:
+			# Use the IAccessible for the active TeamTalk instance's toolbar.
+			NVDAObj = self.getToolBar()
+			if not NVDAObj:
+				# Translators: Reported when status information is unavailable
+				# in the TeamTalk voice conferencing client.
+				ui.message(_("Status unavailable"))
+				return
+		self.NVDAObj = NVDAObj
+		n = NVDAObj.childCount
+		# TeamTalk version-specific toolbar item names by toolbar item count.
+		self.names = []
+		for lst in _msgNames.values():
+			if len(lst) == n:
+				self.names = lst
+				break
+
+	def getToolBar(self):
+		# First lastChild is a Window object, then the client toolbar object.
+		try:
+			tb = api.getForegroundObject().lastChild.lastChild
+		except AttributeError:
+			tb = None
+		return tb
+
+	def name(self, itemID):
+		"""The name of one item.
+		"""
+		if itemID <= 0:
+			return ""
+		# ChildIDs are 1-based, but Python arrays are 0-based.
+		try:
+			name = self.names[itemID - 1]
+		except IndexError:
+			return ""
+		if name == "sep":
+			return ""
+		return name
+
+
+class TTToolBar(IAccessible):
+	def initOverlayClass(self):
+		self.info = TTToolBarInfo(self)
+
+
+class TTToolBarItem(IAccessible):
+	def event_valueChange(self):
+		speech.speakObject(self)
+
+	def _get_name(self):
+		return self.parent.info.name(self.IAccessibleChildID)
+
+
+class AppModule(appModuleHandler.AppModule):
+	def event_NVDAObject_init(self, obj):
+		# For some TeamTalk versions, the incoming chat control does not
+		# return the correct _isWindowUnicode flag.
+		if obj.windowClassName == "RichEdit20A":
+			obj._isWindowUnicode = False
+		# Label channel chat in/out controls.
+		if obj.windowControlID == ID_CHATOUT:
+			# Translators: a label for a read only edit control
+			# in the TeamTalk voice conferencing client where incoming chat
+			# messages are displayed.
+			obj.name = _("Channel messages")
+		elif obj.windowControlID == ID_CHATIN:
+			# Translators: a label for the TeamTalk chat entry field
+			obj.name = _("Type a message to the channel")
```

Comment:
I think this is pretty verbose

```suggestion
			obj.name = _("Channel input")
```